src/main/kotlin/minesweeper/domain/Land.kt

```diff
@@ -1,19 +1,49 @@
 package minesweeper.domain
 
 import minesweeper.domain.tile.Marking
+import minesweeper.domain.tile.Marking.Companion.toMarkingAsInt
+import minesweeper.domain.tile.SurroundingTiles
 import minesweeper.domain.tile.Tiles
+import minesweeper.domain.tile.pos.Coordinate
 import minesweeper.domain.tile.pos.Position
 
-data class Land(private val width: Position, private val tiles: Tiles) {
+data class Land(private val width: Position, private val height: Position, private val tiles: Tiles) {
+    private var _tiles = tiles
+
+    init {
+        for (x in 0..width.value) for (y in 0..height.value) {
+            val coordinate = Coordinate(Position(x), Position(y))
+            _tiles = _tiles.checkTile(coordinate, toMarkingAsInt(getMineCount(coordinate)))
+        }
+    }
+
     fun getTiles(): List<Marking> {
-        return tiles.getList()
+        return _tiles.getList()
     }
 
     fun getWidth() = width.getCalibratedPosition()
 
+    fun getMineCount(coordinate: Coordinate): Int {
+        var mineCount = 0
+        for (surroundingTiles in SurroundingTiles.values()) {
+            val (x, y) = coordinate.getSurroundTilesCoordinateAsPair(surroundingTiles)
+            if (isExistMine(x, y)) mineCount++
+        }
+        return mineCount
```

Comment:
[count](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/count.html)를 활용하면 체이닝된 함수에서 바로 값을 구할 수 있습니다.  Collection API에 유용한 함수가 많아서 "이런것도 있나?" 싶으면 웬만해서 있더라구요 🙂
```suggestion
        return SurroundingTiles.values().count { surroundingTiles ->
            val (x, y) = coordinate.getSurroundTilesCoordinateAsPair(surroundingTiles)
            isExistMine(x, y)
        }
```

Reply:
오... 꿀팁 감사합니다! 알려주신 코드가 훨씬 가독성이 좋네요! 추후에 비슷한 상황이 생기면 Collection API 살펴보겠습니다 😃 
작은 부분까지 세심하게 살펴봐 주셔서 감사합니다 😊

- https://github.com/next-step/kotlin-minesweeper/commit/ccfc72389164dd5a8d08480d1c4d9e89f60e6dde